### ISSUE TITLE
Use inheritance for device_class

### ISSUE THREAD

**Issue by teddyrendahl** [pcdshub/typhon#158]
We will need a way to use bases when comparing device classes. For instance, if I want all my area detector plugins to use the same template you would currently have to:

```python
TyphonDeviceDisplay['device_class='ophyd.areadetectors.plugins.FilePlugin']
TyphonDeviceDisplay['device_class='ophyd.areadetectors.plugins.HDF5Plugin']
...
```
This is unmaintainable

**Comment by klauer** [comment]
Agreed on the unmaintainability. Need to find a better way to support inheritance here.

In QSS, what happens if the property returns a list? Can you somehow check if the list contains a value?
That is, 
```python
def device_hierarchy(cls):
    return [cls.__name__ for cls in cls.mro()]
```

QSS:
```
TyphonDeviceDisplay['device_hierarchy<some_operator_here>'ophyd.areadetectors.plugins.FilePlugin']
```

Even if it's possible, it still wouldn't be perfect. Hmm...

**Comment by teddyrendahl** [comment]
That is interesting and possible ... Now the only thing would be conflict resolution. Qt will just apply which ever rule it sees first if someone specifies a `device_class` rule and a `device_hierarchy` rule if they are deemed both as "specific"
http://doc.qt.io/qt-5/stylesheet-syntax.html#selector-types

**Comment by klauer** [comment]
Crazy hack of an idea that might actually work?

1. Duplicate the entire class hierarchy * **by name only** * for a given Device. No components, no functionality/methods/attributes, nothing else.
2. Make all Typhon widgets dynamically inherit from this hierarchy
3. Use stylesheet selectors natively

**Comment by teddyrendahl** [comment]
> Make all Typhon widgets dynamically inherit from this hierarchy

@klauer What does this step look like in code? The class will have already been instantiated how do you hack the `mro` post instance creation?

**Comment by klauer** [comment]
Good question, @teddyrendahl. I had something in mind when I had originally suggested this, but have apparently long forgotten... Will have to circle back.

**Comment by hhslepicka** [comment]
For the `QStringList` property idea you can do the following as a test:

Add a QLabel to the Designer... create a DynamicProperty with the type `QStringList` with options such as `foo` and `bar` (not original, I know)...

At the stylesheet you can do:
```css
QLabel[prop~='foo']{
    background-color: red;
}
```

The `~=` will check if the QStringList `prop` contains `foo`.

**Comment by teddyrendahl** [comment]
Capturing offline discussion:

Create a `QStringList` as @hhslepicka mentioned above called `mro` which is the `.mro()` call from the device. The issue is then resolving conflicts the example we've been running with is you want a generic `EpicsMotor` screen but then a modified screen for your specific controller type this would look like:

```css
TyphonDeviceDisplay[mro~='ophyd.EpicsMotor'] {}
TyphonDeviceDisplay[device_class='pcdsdsevices.Beckhoff'] {}
```
The issue is that these two conflict in QSS land both will be true for a `Beckhoff`. QSS states that the most specific rule (most requirements) will be applied and if two are of equivalent specificity the second one in the file will be the lasting one. Now we can get around this by being ultra specific and generating a rule that has the entire `mro` in it?
```
TyphonDeviceDisplay[mro~='Beckhoff'][mro~='EpicsMotor][mro~='Device]...
```
This will be both specific enough to ignore rules from parent classes that conflict and still apply to subclasses. The real drawback is this becomes unwritable for a human. I therefore propose the creation of a helper function that would write and append this to the stylesheet for you.

Thoughts? @ZLLentz @klauer

**Comment by hhslepicka** [comment]
@teddyrendahl I think that a "stylesheet editor" would solve this issue in a more user-friendly way but how often will people customize the most specific part?
Also, the creation of stylesheets is not a trivial task. People must understand how CSS/QSS works, the selectors and how conflicts are solved by the QSS engine...

**Comment by teddyrendahl** [comment]
I think the plan is just to create the helper functions and properties in this repository.

In https://github.com/pcdshub/lucid,  I will then make some sort of a configuration file that lives on our NFS system and handles reading and setting up the stylesheet for users:

```yaml
ophyd.EpicsMotor:
    - embedded_screen: my_motor.ui
```

**Comment by hhslepicka** [comment]
Awesome!
